# **KELOMPOK 6 - SISTEM INFORMASI '2024 (*OCEAN PLASTIC POLLUTION*)**

Daftar Nama Anggota:
- Zyrus Alfredo Randan Malinggato (2409116120)
- Jen Agresia Misti (2409116007)
- Raihan Fariz Novanto (2409116083)

Link sumber dataset (*Plastic Pollution* - By: Mohamadreza Momeni): https://www.kaggle.com/datasets/imtkaggleteam/plastic-pollution

# **IMPORT LIBRARY**

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# **BUSINESS UNDERSTANDING**

## **Business Background**

**Latar Belakang Bisnis**

Polusi plastik laut merupakan salah satu permasalahan lingkungan global yang terus meningkat setiap tahunnya. Limbah plastik yang masuk ke lautan dapat berasal dari aktivitas rumah tangga, industri, perikanan, hingga sektor pariwisata. Akumulasi sampah plastik di laut memberikan dampak serius terhadap ekosistem laut, kesehatan manusia, dan aktivitas ekonomi berbasis kelautan.

Saat ini, data mengenai polusi plastik laut tersedia dalam jumlah besar, namun masih tersebar dan sulit dianalisis secara efektif. Kurangnya sistem monitoring yang terintegrasi menyebabkan proses identifikasi wilayah berisiko tinggi dan pengambilan keputusan mitigasi menjadi kurang optimal.

Oleh karena itu, dibutuhkan sebuah sistem berbasis Business Intelligence yang mampu:

- mengintegrasikan data polusi plastik laut,
- menyajikan visualisasi data secara interaktif,
- menghasilkan insight analitis,
- serta mendukung pengambilan keputusan berbasis data.

## **Business Problem**

**Permasalahan Bisnis**

Berikut beberapa permasalahan yang ingin diselesaikan:

a. Permasalahan
- Data polusi laut sulit dianalisis
- Tidak ada monitoring terpusat
- Analisis masih manual
- Tidak ada sistem prioritas risiko
- Volume data besar

b. Dampak
- Informasi tidak efektif
- Sulit melihat tren global
- Pengambilan keputusan lambat
- Sulit menentukan wilayah kritis
- Sulit dipahami tanpa visualisasi

## **Business Objective**

**Tujuan Bisnis**

Tujuan utama proyek ini adalah membangun sistem Business Intelligence yang dapat membantu proses monitoring dan analisis polusi plastik laut secara efektif.

**Objective Utama**

Sistem Diharapkan Mampu:
- Menyediakan informasi mengenai kondisi polusi plastik laut.
- Menampilkan tren polusi berdasarkan waktu dan wilayah.
- Mengidentifikasi negara/wilayah dengan tingkat polusi tinggi.
- Menyediakan dashboard monitoring berbasis MIS.
- Mendukung pengambilan keputusan melalui DSS.
- Menghasilkan insight yang dapat digunakan untuk mitigasi lingkungan.

## **Project Scope**

Sistem Akan Mencakup:

A. Management Information System (MIS)
Fungsi:
- Monitoring data polusi laut
- Visualisasi data
- Reporting
- KPI Monitoring

B. Decision Support System (DSS)
Fungsi:
- Risk Scroring
- Prioritas Wilayah
- Rekomendasi Mitigasi
- Analisis Resiko

C. Business Intelligence
Fungsi:
- Data Analytics
- Trend Analysis
- Comparative Analysis
- Insight Generation

## **Assess Situation**

Analisis situasi saat ini:
1. Data tersedia tetapi tidak terintegrasi
2. Sulit melihat tren
3. Belum ada prioritas risiko
4. Pengolahan data masih manual

Solusi yang ditawarkan:
- Mengintegrasikan dataset polusi plastik laut
- Melakukan ETL dan data warehouse
- Menyediakan dashboard interaktif
- Menghasilkan analisis BI
- Mendukung DSS berbasis risk scoring

## **Analytic Goals**

Analisis yang akan dilakukan, sebagai berikut:
- Trend Anlysis
- Country Analysis
- Plastic Composition Analysis
- Risk Analysis
- Comparative Analysis

## **Project Flow**

- Business Understanding
- Data Understanding
- Data Cleaning
- ETL Process
- Data Warehouse
- SQL Analysis
- BI Dashboard
- DSS Recommendation
- Export Data
- Github

# **Data Understanding**

## **Memuat Dataset (Raw)**

In [ ]:
df1 = pd.read_csv('/content/1- global-plastics-production.csv')
df2 = pd.read_csv('/content/2- share-of-global-plastic-waste-emitted-to-the-ocean.csv')
df3 = pd.read_csv('/content/3- share-plastic-fate.csv')
df4 = pd.read_csv('/content/4- mismanaged-plastic-waste-per-capita.csv')

In [ ]:
df1.head()
df1.info()

df2.head()
df2.info()

df3.head()
df3.info()

df4.head()
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69 entries, 0 to 68
Data columns (total 4 columns):
 #   Column                                           Non-Null Count  Dtype 
---  ------                                           --------------  ----- 
 0   Entity                                           69 non-null     object
 1   Code                                             69 non-null     object
 2   Year                                             69 non-null     int64 
 3   Annual plastic production between 1950 and 2019  69 non-null     int64 
dtypes: int64(2), object(2)
memory usage: 2.3+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 170 entries, 0 to 169
Data columns (total 4 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   Entity                                     170 non-null    object 
 1   Code                                       159 non-null    ob

1. Insight Dataset 1 — global-plastics-production

    Dataset global-plastics-production terdiri dari 69 baris dan 4 kolom yang berisi informasi mengenai produksi plastik tahunan global dari tahun 1950 hingga 2019. Seluruh data memiliki nilai non-null sehingga kualitas data pada dataset ini tergolong baik dan siap digunakan untuk analisis tren produksi plastik. Dataset ini sangat cocok digunakan untuk analisis pertumbuhan produksi plastik dunia dan dapat menjadi indikator utama dalam melihat hubungan antara peningkatan produksi plastik dengan tingkat pencemaran laut.

2. Insight Dataset 2 — share-of-global-plastic-waste-emitted-to-the-ocean

    Dataset share-of-global-plastic-waste-emitted-to-the-ocean terdiri dari 170 baris dan 4 kolom yang berisi persentase kontribusi sampah plastik global yang masuk ke laut berdasarkan negara atau wilayah tertentu. Dataset ini memiliki 11 missing value pada kolom Code, yang kemungkinan disebabkan oleh adanya data agregasi wilayah seperti “World”, “Asia”, atau region lainnya. Dataset ini menjadi dataset utama dalam sistem karena secara langsung merepresentasikan tingkat pencemaran plastik laut dan sangat relevan untuk analisis Business Intelligence, monitoring lingkungan, serta Decision Support System (DSS).

3. Insight Dataset 3 — share-plastic-fate

    Dataset share-plastic-fate terdiri dari 200 baris dan 7 kolom yang menjelaskan distribusi pengelolaan sampah plastik, seperti tingkat daur ulang (recycled), pembakaran (incinerated), sampah tidak terkelola (mismanaged), dan sampah yang ditimbun (landfilled). Dataset ini memiliki jumlah missing value yang cukup besar pada kolom Code, yaitu sebanyak 120 data, yang mengindikasikan adanya data regional atau agregasi global. Dataset ini sangat penting untuk analisis keberlanjutan (sustainability analysis) karena mampu memberikan gambaran mengenai efektivitas pengelolaan sampah plastik di berbagai wilayah.

4. Insight Dataset 4 — mismanaged-plastic-waste-per-capita

    Dataset mismanaged-plastic-waste-per-capita terdiri dari 165 baris dan 4 kolom yang berisi informasi jumlah sampah plastik yang tidak terkelola per kapita dalam satuan kilogram per tahun. Dataset ini memiliki 5 missing value pada kolom Code, namun secara keseluruhan kualitas data masih tergolong baik. Dataset ini sangat cocok digunakan sebagai dasar perhitungan tingkat risiko pencemaran lingkungan karena mampu menunjukkan kontribusi sampah plastik per individu pada masing-masing negara atau wilayah.

## **Informasi Statistik Deskriptif**

### **1. global-plastics-production**

In [ ]:
df1.describe()

,Year,Annual plastic production between 1950 and 2019
count,69.000000,6.900000e+01
mean,1984.652174,1.375638e+08
std,20.460023,1.320418e+08
min,1950.000000,2.000000e+06
25%,1967.000000,2.300000e+07
50%,1985.000000,9.000000e+07
75%,2002.000000,2.310000e+08
max,2019.000000,4.597460e+08


Hasil statistik deskriptif pada dataset *global-plastics-production* menunjukkan bahwa data produksi plastik global mencakup periode tahun 1950 hingga 2019 dengan total 69 data tahunan. Rata-rata produksi plastik global mencapai sekitar 137,5 juta ton per tahun dengan nilai maksimum sebesar 459,7 juta ton dan nilai minimum sebesar 2 juta ton. Perbedaan yang sangat besar antara nilai minimum dan maksimum menunjukkan adanya peningkatan produksi plastik global yang sangat signifikan selama beberapa dekade terakhir.

Nilai standar deviasi yang cukup tinggi, yaitu sekitar 132 juta ton, menunjukkan bahwa persebaran data produksi plastik sangat luas dan tidak merata. Hal ini mengindikasikan adanya pertumbuhan produksi plastik yang terus meningkat dari tahun ke tahun.

Berdasarkan nilai kuartil, sebesar 25% data produksi plastik berada di bawah 23 juta ton, sedangkan 75% data berada di bawah 231 juta ton. Median sebesar 90 juta ton menunjukkan bahwa sebagian besar produksi plastik global masih berada di bawah rata-rata, yang mengindikasikan adanya lonjakan produksi yang sangat tinggi pada tahun-tahun terbaru.

Secara keseluruhan, hasil statistik deskriptif ini menunjukkan bahwa produksi plastik global mengalami tren peningkatan yang sangat signifikan dan berpotensi menjadi salah satu faktor utama meningkatnya pencemaran plastik laut secara global.


### **2. share-of-global-plastic-waste-emitted-to-the-ocean**

In [ ]:
df2.describe()

,Year,Share of global plastics emitted to ocean
count,170.0,170.000000
mean,2019.0,1.177539
std,0.0,6.931025
min,2019.0,0.000000
25%,2019.0,0.000970
50%,2019.0,0.025473
75%,2019.0,0.212005
max,2019.0,80.993570


Hasil statistik deskriptif pada dataset *share-of-global-plastic-waste-emitted-to-the-ocean* menunjukkan bahwa seluruh data berasal dari tahun 2019 dengan total 170 data negara atau wilayah. Dataset ini berisi persentase kontribusi masing-masing negara terhadap total sampah plastik global yang masuk ke laut.

Nilai rata-rata kontribusi polusi plastik laut sebesar 1,18%, sedangkan nilai median hanya sebesar 0,025%. Perbedaan yang cukup besar antara rata-rata dan median menunjukkan bahwa distribusi data tidak merata dan terdapat beberapa negara dengan kontribusi polusi yang sangat tinggi dibandingkan negara lainnya.

Nilai maksimum mencapai sekitar 80,99%, sedangkan nilai minimum bernilai 0%. Hal ini mengindikasikan adanya dominasi kontribusi pencemaran laut oleh negara tertentu, sementara sebagian besar negara memiliki kontribusi yang relatif kecil terhadap total polusi plastik global.

Standar deviasi sebesar 6,93 menunjukkan tingkat variasi data yang cukup tinggi antar negara atau wilayah. Selain itu, nilai kuartil menunjukkan bahwa 75% negara memiliki kontribusi polusi di bawah 0,212%, yang berarti sebagian besar negara sebenarnya memiliki tingkat kontribusi yang rendah, namun terdapat outlier yang sangat dominan.

Secara keseluruhan, hasil statistik deskriptif ini menunjukkan bahwa pencemaran plastik laut global cenderung terpusat pada beberapa negara tertentu, sehingga analisis lebih lanjut sangat penting untuk mengidentifikasi wilayah prioritas dalam mitigasi pencemaran laut.


### **3. share-plastic-fate**

In [ ]:
df3.describe()

,Year,Share of waste recycled from total regional waste,Share of waste incinerated from total regional waste,Share of littered and mismanaged from total regional waste,Share of waste landfilled from total regional waste
count,200.000000,200.000000,200.000000,200.000000,200.000000
mean,2009.500000,6.364439,10.541669,32.703471,50.390421
std,5.780751,2.587257,9.132355,20.154271,16.346958
min,2000.000000,2.345074,0.000000,3.552523,25.098991
25%,2004.750000,4.362588,1.128528,10.955612,33.711067
50%,2009.500000,5.716396,11.009694,32.513705,51.571503
75%,2014.250000,7.920186,16.964613,45.841763,60.648984
max,2019.000000,13.333093,37.823994,69.322820,75.051970


Hasil statistik deskriptif pada dataset *share-plastic-fate* menunjukkan bahwa data mencakup periode tahun 2000 hingga 2019 dengan total 200 data wilayah atau negara. Dataset ini memberikan gambaran mengenai distribusi pengelolaan sampah plastik global berdasarkan empat kategori utama, yaitu *recycled*, *incinerated*, *mismanaged*, dan *landfilled*.

Rata-rata tingkat sampah plastik yang didaur ulang (*recycled*) hanya sebesar 6,36%, dengan nilai maksimum sekitar 13,33%. Hal ini menunjukkan bahwa tingkat daur ulang sampah plastik secara global masih relatif rendah. Sebaliknya, rata-rata sampah plastik yang ditimbun di landfill (*landfilled*) mencapai 50,39%, menjadikannya metode pengelolaan sampah yang paling dominan.

Selain itu, rata-rata sampah plastik yang termasuk kategori littered dan mismanaged mencapai 32,70%, yang menunjukkan bahwa masih terdapat proporsi besar sampah plastik yang tidak terkelola dengan baik dan berpotensi mencemari lingkungan, termasuk lautan.

Nilai standar deviasi yang cukup tinggi pada kategori *mismanaged waste* dan *landfilled waste* menunjukkan adanya perbedaan signifikan antar wilayah dalam pengelolaan sampah plastik. Beberapa wilayah memiliki sistem pengelolaan sampah yang lebih baik, sedangkan wilayah lainnya masih memiliki tingkat sampah tidak terkelola yang tinggi.

Secara keseluruhan, hasil statistik deskriptif ini menunjukkan bahwa pengelolaan sampah plastik global masih didominasi oleh metode landfill dan masih rendahnya tingkat daur ulang. Tingginya proporsi sampah plastik yang tidak terkelola dengan baik menjadi salah satu faktor penting yang berkontribusi terhadap peningkatan pencemaran plastik laut.

### **4. mismanaged-plastic-waste-per-capita**

In [ ]:
df4.describe()

,Year,Mismanaged plastic waste per capita (kg per year)
count,165.0,165.000000
mean,2019.0,8.564164
std,0.0,10.485455
min,2019.0,0.002720
25%,2019.0,0.500787
50%,2019.0,5.981680
75%,2019.0,12.510074
max,2019.0,69.515864


Hasil statistik deskriptif pada dataset *mismanaged-plastic-waste-per-capita* menunjukkan bahwa seluruh data berasal dari tahun 2019 dengan total 165 data negara atau wilayah. Dataset ini mengukur jumlah sampah plastik yang tidak terkelola per kapita dalam satuan kilogram per tahun.

Rata-rata sampah plastik tidak terkelola per kapita sebesar 8,56 kg per tahun, sedangkan nilai median sebesar 5,98 kg per tahun. Perbedaan antara rata-rata dan median menunjukkan bahwa terdapat beberapa negara dengan tingkat sampah tidak terkelola yang sangat tinggi sehingga memengaruhi rata-rata keseluruhan data.

Nilai maksimum mencapai sekitar 69,52 kg per kapita per tahun, sedangkan nilai minimum hanya sekitar 0,003 kg per kapita per tahun. Perbedaan yang sangat besar antara nilai minimum dan maksimum menunjukkan adanya ketimpangan tingkat pengelolaan sampah plastik antar negara atau wilayah.

Standar deviasi sebesar 10,49 menunjukkan bahwa distribusi data cukup bervariasi dan terdapat perbedaan signifikan dalam tingkat pengelolaan sampah plastik per individu. Selain itu, nilai kuartil menunjukkan bahwa 75% negara memiliki tingkat sampah tidak terkelola di bawah 12,51 kg per kapita per tahun, sementara sebagian kecil negara memiliki nilai yang jauh lebih tinggi dan berpotensi menjadi wilayah prioritas mitigasi.

Secara keseluruhan, hasil statistik deskriptif ini menunjukkan bahwa masih terdapat banyak negara dengan tingkat sampah plastik tidak terkelola yang tinggi, sehingga diperlukan analisis lebih lanjut untuk menentukan wilayah dengan tingkat risiko pencemaran lingkungan tertinggi.

# **Data Cleaning & Standardization**

## **Rename Column**

In [ ]:
df1 = df1.rename(columns={
    'Entity': 'entity',
    'Code': 'code',
    'Year': 'year',
    'Annual plastic production between 1950 and 2019': 'plastic_production'
})

df2 = df2.rename(columns={
    'Entity': 'entity',
    'Code': 'code',
    'Year': 'year',
    'Share of global plastics emitted to ocean': 'ocean_pollution_share'
})

df3 = df3.rename(columns={
    'Entity': 'entity',
    'Code': 'code',
    'Year': 'year',
    'Share of waste recycled from total regional waste': 'recycled_share',
    'Share of waste incinerated from total regional waste': 'incinerated_share',
    'Share of littered and mismanaged from total regional waste': 'mismanaged_share',
    'Share of waste landfilled from total regional waste': 'landfilled_share'
})

df4 = df4.rename(columns={
    'Entity': 'entity',
    'Code': 'code',
    'Year': 'year',
    'Mismanaged plastic waste per capita (kg per year)': 'mismanaged_per_capita'
})

In [ ]:
print(df1.columns)
print(df2.columns)
print(df3.columns)
print(df4.columns)

Index(['entity', 'code', 'year', 'plastic_production'], dtype='object')
Index(['entity', 'code', 'year', 'ocean_pollution_share'], dtype='object')
Index(['entity', 'code', 'year', 'recycled_share', 'incinerated_share',
       'mismanaged_share', 'landfilled_share'],
      dtype='object')
Index(['entity', 'code', 'year', 'mismanaged_per_capita'], dtype='object')


## **Cek Missing Value**

In [ ]:
print("Missing value df1:")
print(df1.isnull().sum())

print("\nMissing value df2:")
print(df2.isnull().sum())

print("\nMissing value df3:")
print(df3.isnull().sum())

print("\nMissing value df4:")
print(df4.isnull().sum())

Missing value df1:
entity                0
code                  0
year                  0
plastic_production    0
dtype: int64

Missing value df2:
entity                    0
code                     11
year                      0
ocean_pollution_share     0
dtype: int64

Missing value df3:
entity                 0
code                 120
year                   0
recycled_share         0
incinerated_share      0
mismanaged_share       0
landfilled_share       0
dtype: int64

Missing value df4:
entity                   0
code                     5
year                     0
mismanaged_per_capita    0
dtype: int64


### **Menangani Missing Values**

In [ ]:
df1['code'] = df1['code'].fillna('UNKNOWN')
df2['code'] = df2['code'].fillna('UNKNOWN')
df3['code'] = df3['code'].fillna('UNKNOWN')
df4['code'] = df4['code'].fillna('UNKNOWN')

In [ ]:
print(df1.isnull().sum())
print(df2.isnull().sum())
print(df3.isnull().sum())
print(df4.isnull().sum())

entity                0
code                  0
year                  0
plastic_production    0
dtype: int64
entity                   0
code                     0
year                     0
ocean_pollution_share    0
dtype: int64
entity               0
code                 0
year                 0
recycled_share       0
incinerated_share    0
mismanaged_share     0
landfilled_share     0
dtype: int64
entity                   0
code                     0
year                     0
mismanaged_per_capita    0
dtype: int64


## **Cek Duplicate**

In [ ]:
print("Duplicate df1:", df1.duplicated().sum())
print("Duplicate df2:", df2.duplicated().sum())
print("Duplicate df3:", df3.duplicated().sum())
print("Duplicate df4:", df4.duplicated().sum())

Duplicate df1: 0
Duplicate df2: 0
Duplicate df3: 0
Duplicate df4: 0


## **Standardisasi Teks**

In [ ]:
for df in [df1, df2, df3, df4]:
    df['entity'] = df['entity'].str.strip()
    df['code'] = df['code'].str.strip()

## **Cek Tahun Tiap Dataset**

In [ ]:
print("df1 year range:", df1['year'].min(), "-", df1['year'].max())
print("df2 year range:", df2['year'].min(), "-", df2['year'].max())
print("df3 year range:", df3['year'].min(), "-", df3['year'].max())
print("df4 year range:", df4['year'].min(), "-", df4['year'].max())

df1 year range: 1950 - 2019
df2 year range: 2019 - 2019
df3 year range: 2000 - 2019
df4 year range: 2019 - 2019


## **Simpan Clean Dataset**

In [ ]:
df1.to_csv('clean_global_plastics_production.csv', index=False)
df2.to_csv('clean_ocean_pollution_share.csv', index=False)
df3.to_csv('clean_plastic_fate.csv', index=False)
df4.to_csv('clean_mismanaged_per_capita.csv', index=False)

# **Data Integration**

Tahap data integration dilakukan dengan menggabungkan seluruh dataset berdasarkan kolom entity dan year untuk membentuk satu integrated analytical dataset. Proses integrasi ini memungkinkan seluruh informasi terkait produksi plastik, kontribusi pencemaran laut, pengelolaan sampah plastik, dan tingkat sampah tidak terkelola per kapita berada dalam satu struktur data yang terintegrasi.

Hasil integrasi data menghasilkan dataset terpadu yang dapat digunakan untuk kebutuhan Business Intelligence, analisis SQL, visualisasi Tableau, serta pengembangan Decision Support System (DSS). Selain itu, integrated dataset ini juga menjadi dasar dalam pembangunan Data Warehouse dan dashboard monitoring polusi plastik laut.

## **Memuat Clean Dataset**

In [ ]:
df1 = pd.read_csv('/content/clean_global_plastics_production.csv')
df2 = pd.read_csv('/content/clean_ocean_pollution_share.csv')
df3 = pd.read_csv('/content/clean_plastic_fate.csv')
df4 = pd.read_csv('/content/clean_mismanaged_per_capita.csv')

## **Cek Struktur Dataset Clean**

In [ ]:
print(df1.columns)
print(df2.columns)
print(df3.columns)
print(df4.columns)

Index(['entity', 'code', 'year', 'plastic_production'], dtype='object')
Index(['entity', 'code', 'year', 'ocean_pollution_share'], dtype='object')
Index(['entity', 'code', 'year', 'recycled_share', 'incinerated_share',
       'mismanaged_share', 'landfilled_share'],
      dtype='object')
Index(['entity', 'code', 'year', 'mismanaged_per_capita'], dtype='object')


## **Cek Ukuran Dataset**

In [ ]:
print(df1.shape)
print(df2.shape)
print(df3.shape)
print(df4.shape)

(69, 4)
(170, 4)
(200, 7)
(165, 4)


## **Merge Dataset 1 & Dataset 2**

In [ ]:
merged_df = pd.merge(
    df1,
    df2,
    on=['entity', 'year'],
    how='outer'
)

### **Cek Hasil Merge**

In [ ]:
merged_df.head()

,entity,code_x,year,plastic_production,code_y,ocean_pollution_share
0,Africa,NaN,2019,NaN,UNKNOWN,7.989317
1,Albania,NaN,2019,NaN,ALB,0.159782
2,Algeria,NaN,2019,NaN,DZA,0.589510
3,Angola,NaN,2019,NaN,AGO,0.087804
4,Antigua and Barbuda,NaN,2019,NaN,ATG,0.000204


In [ ]:
merged_df.shape

(239, 6)

## **Merge Dengan Dataset 3**

In [ ]:
merged_df = pd.merge(
    merged_df,
    df3,
    on=['entity', 'year'],
    how='outer'
)

## **Merge Dengan Dataset 4**

In [ ]:
print(merged_df.columns)

Index(['entity', 'code_x', 'year', 'plastic_production', 'code_y',
       'ocean_pollution_share', 'code', 'recycled_share', 'incinerated_share',
       'mismanaged_share', 'landfilled_share'],
      dtype='object')


In [ ]:
code_cols = [col for col in merged_df.columns if 'code' in col]

merged_df['code'] = merged_df[code_cols].bfill(axis=1).iloc[:, 0]

merged_df = merged_df.drop(columns=[col for col in code_cols if col != 'code'])

In [ ]:
print(merged_df.columns)

Index(['entity', 'year', 'plastic_production', 'ocean_pollution_share', 'code',
       'recycled_share', 'incinerated_share', 'mismanaged_share',
       'landfilled_share'],
      dtype='object')


In [ ]:
merged_df = pd.merge(
    merged_df,
    df4,
    on=['entity', 'year'],
    how='outer',
    suffixes=('', '_df4')
)

In [ ]:
merged_df['code'] = merged_df['code'].fillna(merged_df['code_df4'])

merged_df = merged_df.drop(columns=['code_df4'])

## **Validasi Hasil Merge**

### **Lihat Struktur Dataset Final**

In [ ]:
merged_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 414 entries, 0 to 413
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   entity                 414 non-null    object 
 1   year                   414 non-null    int64  
 2   plastic_production     69 non-null     float64
 3   ocean_pollution_share  170 non-null    float64
 4   code                   414 non-null    object 
 5   recycled_share         200 non-null    float64
 6   incinerated_share      200 non-null    float64
 7   mismanaged_share       200 non-null    float64
 8   landfilled_share       200 non-null    float64
 9   mismanaged_per_capita  165 non-null    float64
dtypes: float64(7), int64(1), object(2)
memory usage: 32.5+ KB


### **Lihat Ukuran Dataset Final**

In [ ]:
merged_df.shape

(414, 10)

### **Preview Dataset**

In [ ]:
merged_df.head(10)

,entity,year,plastic_production,ocean_pollution_share,code,recycled_share,incinerated_share,mismanaged_share,landfilled_share,mismanaged_per_capita
0,Africa,2019,NaN,7.989317,UNKNOWN,NaN,NaN,NaN,NaN,10.465928
1,Albania,2019,NaN,0.159782,ALB,NaN,NaN,NaN,NaN,24.239153
2,Algeria,2019,NaN,0.589510,DZA,NaN,NaN,NaN,NaN,17.758995
3,Americas (excl. USA),2000,NaN,NaN,UNKNOWN,4.721963,1.220057,32.702644,61.355328,NaN
4,Americas (excl. USA),2001,NaN,NaN,UNKNOWN,4.972505,1.208672,32.651270,61.167557,NaN
5,Americas (excl. USA),2002,NaN,NaN,UNKNOWN,5.227469,1.197566,32.610554,60.964405,NaN
6,Americas (excl. USA),2003,NaN,NaN,UNKNOWN,5.482511,1.186352,32.546440,60.784700,NaN
7,Americas (excl. USA),2004,NaN,NaN,UNKNOWN,5.740007,1.175283,32.480970,60.603745,NaN
8,Americas (excl. USA),2005,NaN,NaN,UNKNOWN,5.993266,1.164319,32.386340,60.456074,NaN
9,Americas (excl. USA),2006,NaN,NaN,UNKNOWN,6.248887,1.153630,32.298714,60.298763,NaN


## **Validation**

### **Cek Missing Values**

In [ ]:
merged_df.isnull().sum()

,0
entity,0
year,0
plastic_production,345
ocean_pollution_share,244
code,0
recycled_share,214
incinerated_share,214
mismanaged_share,214
landfilled_share,214
mismanaged_per_capita,249


### **Cek Duplicate**

In [ ]:
merged_df.duplicated().sum()

np.int64(0)

In [ ]:
merged_df = merged_df.drop_duplicates()

### **Cek Jumlah Entity**

In [ ]:
merged_df['entity'].nunique()

175

### **Cek Range Tahun**

In [ ]:
merged_df['year'].min(), merged_df['year'].max()

(1950, 2019)

## **Cek Integrated Dataset Final**

In [ ]:
merged_df.head(20)

,entity,year,plastic_production,ocean_pollution_share,code,recycled_share,incinerated_share,mismanaged_share,landfilled_share,mismanaged_per_capita
0,Africa,2019,NaN,7.989317,UNKNOWN,NaN,NaN,NaN,NaN,10.465928
1,Albania,2019,NaN,0.159782,ALB,NaN,NaN,NaN,NaN,24.239153
2,Algeria,2019,NaN,0.589510,DZA,NaN,NaN,NaN,NaN,17.758995
3,Americas (excl. USA),2000,NaN,NaN,UNKNOWN,4.721963,1.220057,32.702644,61.355328,NaN
4,Americas (excl. USA),2001,NaN,NaN,UNKNOWN,4.972505,1.208672,32.651270,61.167557,NaN
5,Americas (excl. USA),2002,NaN,NaN,UNKNOWN,5.227469,1.197566,32.610554,60.964405,NaN
6,Americas (excl. USA),2003,NaN,NaN,UNKNOWN,5.482511,1.186352,32.546440,60.784700,NaN
7,Americas (excl. USA),2004,NaN,NaN,UNKNOWN,5.740007,1.175283,32.480970,60.603745,NaN
8,Americas (excl. USA),2005,NaN,NaN,UNKNOWN,5.993266,1.164319,32.386340,60.456074,NaN
9,Americas (excl. USA),2006,NaN,NaN,UNKNOWN,6.248887,1.153630,32.298714,60.298763,NaN


## **Simpan Integrated Dataset**

In [ ]:
merged_df.to_csv(
    'integrated_ocean_plastic_dataset.csv',
    index=False
)

# **Feature Engineering**

## **Memuat Data**

In [ ]:
dss_df = pd.read_csv('/content/integrated_ocean_plastic_dataset.csv')

## **Filter Data 2019**

In [ ]:
dss_df = merged_df[merged_df['year'] == 2019].copy()

In [ ]:
dss_df.shape

(175, 10)

In [ ]:
dss_df.head()

,entity,year,plastic_production,ocean_pollution_share,code,recycled_share,incinerated_share,mismanaged_share,landfilled_share,mismanaged_per_capita
0,Africa,2019,NaN,7.989317,UNKNOWN,NaN,NaN,NaN,NaN,10.465928
1,Albania,2019,NaN,0.159782,ALB,NaN,NaN,NaN,NaN,24.239153
2,Algeria,2019,NaN,0.589510,DZA,NaN,NaN,NaN,NaN,17.758995
22,Americas (excl. USA),2019,NaN,NaN,UNKNOWN,9.554245,1.15926,29.180883,60.10561,NaN
23,Angola,2019,NaN,0.087804,AGO,NaN,NaN,NaN,NaN,7.445279


## **Missing Value DSS Dataset**

In [ ]:
dss_df.isnull().sum()

,0
entity,0
year,0
plastic_production,174
ocean_pollution_share,5
code,0
recycled_share,165
incinerated_share,165
mismanaged_share,165
landfilled_share,165
mismanaged_per_capita,10


### **Handling Missing Values DSS**

In [ ]:
dss_columns = [
    'ocean_pollution_share',
    'mismanaged_share',
    'recycled_share',
    'mismanaged_per_capita'
]

for col in dss_columns:
    dss_df[col] = dss_df[col].fillna(
        dss_df[col].median()
    )

## **Normalisasi Data**

### **Import Library**

In [ ]:
from sklearn.preprocessing import MinMaxScaler

In [ ]:
features = [
    'ocean_pollution_share',
    'mismanaged_share',
    'recycled_share',
    'mismanaged_per_capita'
]

In [ ]:
scaler = MinMaxScaler()

dss_df[features] = scaler.fit_transform(
    dss_df[features]
)

## **Membuat Risk Score**

### **Implementasi Python**

In [ ]:
dss_df['risk_score'] = (
    (0.4 * dss_df['ocean_pollution_share']) +
    (0.3 * dss_df['mismanaged_share']) +
    (0.2 * dss_df['mismanaged_per_capita']) -
    (0.1 * dss_df['recycled_share'])
)

## **Ubah Ke Skala 0-100**

In [ ]:
dss_df['risk_score'] = dss_df['risk_score'] * 100

## **Membuat Risk Category**

In [ ]:
def categorize_risk(score):
    if score >= 80:
        return 'Critical'
    elif score >= 60:
        return 'High'
    elif score >= 40:
        return 'Medium'
    else:
        return 'Low'

dss_df['risk_category'] = dss_df['risk_score'].apply(categorize_risk)

## **Cek Hasil DSS**

In [ ]:
dss_df[['entity', 'risk_score', 'risk_category']].head(20)

,entity,risk_score,risk_category
0,Africa,14.327665,Low
1,Albania,14.423690,Low
2,Algeria,12.771477,Low
22,Americas (excl. USA),8.807964,Low
23,Angola,9.556287,Low
24,Antigua and Barbuda,9.230672,Low
25,Argentina,10.572190,Low
26,Asia,49.864499,Medium
46,Asia (excl. China and India),12.448538,Low
47,Australia,7.432532,Low


## **Top High Risk Countries**

In [ ]:
dss_df.sort_values(
    by='risk_score',
    ascending=False
)[['entity', 'risk_score', 'risk_category']].head(10)

,entity,risk_score,risk_category
26,Asia,49.864499,Medium
255,Philippines,36.051809,Low
303,Sub-Saharan Africa,30.341569,Low
86,Comoros,27.371585,Low
313,Trinidad and Tobago,22.634896,Low
164,India,20.425784,Low
305,Suriname,18.811947,Low
214,Middle East & North Africa,18.754255,Low
186,Malaysia,18.390881,Low
413,Zimbabwe,17.682289,Low


## **Simpan DSS Dataset**

In [ ]:
dss_df.to_csv(
    'ocean_plastic_dss_dataset.csv',
    index=False
)

# **Data Warehouse Design**

Data Warehouse dirancang menggunakan pendekatan star schema yang terdiri dari satu fact table dan beberapa dimension table. Fact table digunakan untuk menyimpan data numerik utama seperti produksi plastik, kontribusi pencemaran laut, tingkat daur ulang, serta risk score hasil DSS. Sementara itu, dimension table digunakan untuk menyimpan informasi deskriptif seperti negara, tahun, dan kategori risiko.

Pendekatan star schema dipilih karena mampu mempermudah proses analisis data, query SQL, serta visualisasi dashboard pada Tableau. Selain itu, struktur ini juga mendukung kebutuhan Business Intelligence dan Decision Support System secara lebih efisien.

## **Load DSS Target**

In [ ]:
dw_df = pd.read_csv('/content/ocean_plastic_dss_dataset.csv')

## **Cek Dataset**

In [ ]:
dw_df.head()

,entity,year,plastic_production,ocean_pollution_share,code,recycled_share,incinerated_share,mismanaged_share,landfilled_share,mismanaged_per_capita,risk_score,risk_category
0,Africa,2019,NaN,0.098641,UNKNOWN,0.489916,NaN,0.409025,NaN,0.150521,14.327665,Low
1,Albania,2019,NaN,0.001973,ALB,0.489916,NaN,0.409025,NaN,0.348660,14.423690,Low
2,Algeria,2019,NaN,0.007278,DZA,0.489916,NaN,0.409025,NaN,0.255438,12.771477,Low
3,Americas (excl. USA),2019,NaN,0.000315,UNKNOWN,0.572732,1.15926,0.426749,60.10561,0.086012,8.807964,Low
4,Angola,2019,NaN,0.001084,AGO,0.489916,NaN,0.409025,NaN,0.107067,9.556287,Low


## **Membuat DIM_COUNTRY**

In [ ]:
dim_country = dw_df[['entity', 'code']].drop_duplicates()

### **Tambahkan ID**

In [ ]:
dim_country['country_id'] = range(1, len(dim_country) + 1)

### **Susun Ulang Kolom**

In [ ]:
dim_country = dim_country[
    ['country_id', 'entity', 'code']
]

In [ ]:
dim_country.head()

,country_id,entity,code
0,1,Africa,UNKNOWN
1,2,Albania,ALB
2,3,Algeria,DZA
3,4,Americas (excl. USA),UNKNOWN
4,5,Angola,AGO


## **Membuat DIM_DATE**

In [ ]:
dim_date = dw_df[['year']].drop_duplicates()

### **Tambahkan ID**

In [ ]:
dim_date['date_id'] = range(1, len(dim_date) + 1)

### **Susun Ulang**

In [ ]:
dim_date = dim_date[
    ['date_id', 'year']
]

## **Membuat DIM_RISK**

In [ ]:
dim_risk = dw_df[['risk_category']].drop_duplicates()

### **Tambahkan Deskripsi**

In [ ]:
risk_desc = {
    'Critical': 'Very High Pollution Risk',
    'High': 'High Pollution Risk',
    'Medium': 'Moderate Pollution Risk',
    'Low': 'Low Pollution Risk'
}

dim_risk['risk_description'] = dim_risk[
    'risk_category'
].map(risk_desc)

In [ ]:
dim_risk.head()

,risk_category,risk_description
0,Low,Low Pollution Risk
7,Medium,Moderate Pollution Risk


## **Membuat Fact Table**

### **Merge country_id**

In [ ]:
fact_table = pd.merge(
    dw_df,
    dim_country,
    on=['entity', 'code'],
    how='left'
)

### **Merge date_id**

In [ ]:
fact_table = pd.merge(
    fact_table,
    dim_date,
    on='year',
    how='left'
)

## **Pilih Kolom Fact Table**

In [ ]:
fact_table = fact_table[
    [
        'country_id',
        'date_id',
        'plastic_production',
        'ocean_pollution_share',
        'recycled_share',
        'incinerated_share',
        'mismanaged_share',
        'landfilled_share',
        'mismanaged_per_capita',
        'risk_score'
    ]
]

## **Preview Fact Table**

In [ ]:
fact_table.head()

,country_id,date_id,plastic_production,ocean_pollution_share,recycled_share,incinerated_share,mismanaged_share,landfilled_share,mismanaged_per_capita,risk_score
0,1,1,NaN,0.098641,0.489916,NaN,0.409025,NaN,0.150521,14.327665
1,2,1,NaN,0.001973,0.489916,NaN,0.409025,NaN,0.348660,14.423690
2,3,1,NaN,0.007278,0.489916,NaN,0.409025,NaN,0.255438,12.771477
3,4,1,NaN,0.000315,0.572732,1.15926,0.426749,60.10561,0.086012,8.807964
4,5,1,NaN,0.001084,0.489916,NaN,0.409025,NaN,0.107067,9.556287


## **Simpan Data Warehouse**

### **Save Dimension**

In [ ]:
dim_country.to_csv('dim_country.csv', index=False)
dim_date.to_csv('dim_date.csv', index=False)
dim_risk.to_csv('dim_risk.csv', index=False)

### **Save Fact Table**

In [ ]:
fact_table.to_csv('fact_ocean_pollution.csv', index=False)

# **SQL Analysis**

Tahap SQL Analysis dilakukan untuk memperoleh insight dari Data Warehouse yang telah dibangun. Analisis dilakukan dengan menggabungkan fact table dan dimension table menggunakan query SQL. Hasil query digunakan untuk mengidentifikasi negara atau wilayah dengan tingkat risiko tertinggi, kontribusi polusi plastik laut terbesar, tingkat sampah tidak terkelola per kapita tertinggi, serta indikator pengelolaan sampah plastik seperti recycling, incineration, mismanaged waste, dan landfill.

SQL Analysis ini menjadi dasar dalam penyusunan dashboard Business Intelligence dan Decision Support System karena menghasilkan informasi yang dapat digunakan untuk monitoring, perbandingan antar wilayah, dan penentuan prioritas mitigasi pencemaran plastik laut.

## **Load Data Warehouse Ke SQLite**

In [ ]:
import sqlite3
import pandas as pd

conn = sqlite3.connect('ocean_plastic_dw.db')

dim_country.to_sql('dim_country', conn, if_exists='replace', index=False)
dim_date.to_sql('dim_date', conn, if_exists='replace', index=False)
dim_risk.to_sql('dim_risk', conn, if_exists='replace', index=False)
fact_table.to_sql('fact_ocean_pollution', conn, if_exists='replace', index=False)

175

## **Cek Tabel di Database**

In [ ]:
query = """
SELECT name
FROM sqlite_master
WHERE type='table';
"""

pd.read_sql_query(query, conn)

,name
0,dim_country
1,dim_date
2,dim_risk
3,fact_ocean_pollution


## **Query 1 - Melihat Data Gabungan**

In [ ]:
query = """
SELECT
    c.entity,
    c.code,
    d.year,
    f.plastic_production,
    f.ocean_pollution_share,
    f.recycled_share,
    f.mismanaged_share,
    f.landfilled_share,
    f.mismanaged_per_capita,
    f.risk_score
FROM fact_ocean_pollution f
JOIN dim_country c ON f.country_id = c.country_id
JOIN dim_date d ON f.date_id = d.date_id;
"""

pd.read_sql_query(query, conn)

,entity,code,year,plastic_production,ocean_pollution_share,recycled_share,mismanaged_share,landfilled_share,mismanaged_per_capita,risk_score
0,Africa,UNKNOWN,2019,NaN,0.098641,0.489916,0.409025,NaN,0.150521,14.327665
1,Albania,ALB,2019,NaN,0.001973,0.489916,0.409025,NaN,0.348660,14.423690
2,Algeria,DZA,2019,NaN,0.007278,0.489916,0.409025,NaN,0.255438,12.771477
3,Americas (excl. USA),UNKNOWN,2019,NaN,0.000315,0.572732,0.426749,60.105610,0.086012,8.807964
4,Angola,AGO,2019,NaN,0.001084,0.489916,0.409025,NaN,0.107067,9.556287
...,...,...,...,...,...,...,...,...,...,...
170,Vietnam,VNM,2019,NaN,0.035574,0.489916,0.409025,NaN,0.165916,12.112871
171,Western Sahara,ESH,2019,NaN,0.000048,0.489916,0.409025,NaN,0.101650,9.406500
172,World,OWID_WRL,2019,459746020.0,0.000315,0.543119,0.314745,49.205143,0.115170,6.327127
173,Yemen,YEM,2019,NaN,0.000318,0.489916,0.409025,NaN,0.143876,10.261817


## **Query 2 — Top 10 Negara/Wilayah dengan Risk Score Tertinggi**

In [ ]:
query = """
SELECT
    c.entity,
    d.year,
    f.risk_score
FROM fact_ocean_pollution f
JOIN dim_country c ON f.country_id = c.country_id
JOIN dim_date d ON f.date_id = d.date_id
ORDER BY f.risk_score DESC
LIMIT 10;
"""

pd.read_sql_query(query, conn)

,entity,year,risk_score
0,Asia,2019,49.864499
1,Philippines,2019,36.051809
2,Sub-Saharan Africa,2019,30.341569
3,Comoros,2019,27.371585
4,Trinidad and Tobago,2019,22.634896
5,India,2019,20.425784
6,Suriname,2019,18.811947
7,Middle East & North Africa,2019,18.754255
8,Malaysia,2019,18.390881
9,Zimbabwe,2019,17.682289


## **Query 3 — Top 10 Kontributor Polusi Plastik Laut**

In [ ]:
query = """
SELECT
    c.entity,
    d.year,
    f.ocean_pollution_share
FROM fact_ocean_pollution f
JOIN dim_country c ON f.country_id = c.country_id
JOIN dim_date d ON f.date_id = d.date_id
ORDER BY f.ocean_pollution_share DESC
LIMIT 10;
"""

pd.read_sql_query(query, conn)

,entity,year,ocean_pollution_share
0,Asia,2019,1.000000
1,Philippines,2019,0.449227
2,India,2019,0.159477
3,Africa,2019,0.098641
4,Malaysia,2019,0.092144
5,China,2019,0.089130
6,Indonesia,2019,0.071011
7,South America,2019,0.068070
8,North America,2019,0.055549
9,Brazil,2019,0.047648


## **Query 4 — Negara/Wilayah dengan Mismanaged Waste Per Capita Tertinggi**

In [ ]:
query = """
SELECT
    c.entity,
    d.year,
    f.mismanaged_per_capita
FROM fact_ocean_pollution f
JOIN dim_country c ON f.country_id = c.country_id
JOIN dim_date d ON f.date_id = d.date_id
ORDER BY f.mismanaged_per_capita DESC
LIMIT 10;
"""

pd.read_sql_query(query, conn)

,entity,year,mismanaged_per_capita
0,Comoros,2019,1.000000
1,Trinidad and Tobago,2019,0.754198
2,Suriname,2019,0.567790
3,Philippines,2019,0.535557
4,Zimbabwe,2019,0.515535
5,Guyana,2019,0.506402
6,Tanzania,2019,0.425644
7,Libya,2019,0.400171
8,Uruguay,2019,0.384828
9,Malaysia,2019,0.366676


## **Query 5 — Rata-Rata Waste Management Indicator**

In [ ]:
query = """
SELECT
    AVG(recycled_share) AS avg_recycled_share,
    AVG(incinerated_share) AS avg_incinerated_share,
    AVG(mismanaged_share) AS avg_mismanaged_share,
    AVG(landfilled_share) AS avg_landfilled_share
FROM fact_ocean_pollution;
"""

pd.read_sql_query(query, conn)

,avg_recycled_share,avg_incinerated_share,avg_mismanaged_share,avg_landfilled_share
0,0.489933,13.609163,0.408726,49.763801


## **Query 6 — Analisis Produksi Plastik per Tahun**

In [ ]:
query = """
SELECT
    d.year,
    SUM(f.plastic_production) AS total_plastic_production
FROM fact_ocean_pollution f
JOIN dim_date d ON f.date_id = d.date_id
GROUP BY d.year
ORDER BY d.year;
"""

pd.read_sql_query(query, conn)

,year,total_plastic_production
0,2019,459746020.0


## **Query 7 — Negara dengan Recycling Rate Tertinggi**

In [ ]:
query = """
SELECT
    c.entity,
    d.year,
    f.recycled_share
FROM fact_ocean_pollution f
JOIN dim_country c ON f.country_id = c.country_id
JOIN dim_date d ON f.date_id = d.date_id
ORDER BY f.recycled_share DESC
LIMIT 10;
"""

pd.read_sql_query(query, conn)

,entity,year,recycled_share
0,India,2019,1.000000
1,China,2019,0.939853
2,Europe,2019,0.892734
3,Americas (excl. USA),2019,0.572732
4,World,2019,0.543119
5,Africa,2019,0.489916
6,Albania,2019,0.489916
7,Algeria,2019,0.489916
8,Angola,2019,0.489916
9,Antigua and Barbuda,2019,0.489916


## **Query 8 — Negara Prioritas Mitigasi**

In [ ]:
query = """
SELECT
    c.entity,
    d.year,
    f.ocean_pollution_share,
    f.mismanaged_share,
    f.mismanaged_per_capita,
    f.recycled_share,
    f.risk_score
FROM fact_ocean_pollution f
JOIN dim_country c ON f.country_id = c.country_id
JOIN dim_date d ON f.date_id = d.date_id
WHERE f.risk_score >= 60
ORDER BY f.risk_score DESC;
"""

pd.read_sql_query(query, conn)

,entity,year,ocean_pollution_share,mismanaged_share,mismanaged_per_capita,recycled_share,risk_score
